File: service/internal/api/v1/article.go
package v1

import (
	"github.com/gin-gonic/gin"
	"github/gin-react-admin/internal/model"
	"github/gin-react-admin/internal/service"
	"github/gin-react-admin/pkg/errcode"
	"github/gin-react-admin/pkg/request"
	"github/gin-react-admin/pkg/response"
	"github/gin-react-admin/utils"
)


// @Tags Article
// @Summary 创建文章
// @Produce  application/json
// @Success 200 {string} string "{state:{code:0, msg:"成功"},"data":{ }}"
// @Router /api/v1/article/create_article [post]
func CreateArticle(c *gin.Context)  {
	var newArticle request.Article
	if err := c.ShouldBindJSON(&newArticle); err != nil {
		res := response.Response{Err: err}
		res.SendInputParameterError(c)
		return
	}
	value, _ := utils.GetHeaderUser(c)
 	ar := model.Article {
			ArticleFileName: newArticle.FileName,
			ArticleContext: newArticle.ArticleContext,
			ArticleAuthor: value.NickName,
			ArticleName: newArticle.FileName,
			ArticleStorageType: newArticle.ArticleStorageType,
			ArticleType: newArticle.ArticleType,
			ArticleTitle: newArticle.ArticleTitle,
			ArticleUrl: newArticle.ArticleUrl,
			UUID: value.UUID,
			ArticleDesc: newArticle.ArticleDesc,
	}
	art, er := service.CreateArticle(&ar, newArticle.ArticleTags)
	if er != nil {
		res := response.Response{
			Err: er,
			State: response.State{Message: "文章创建失败", Code: errcode.CreateError},
		}
		res.SendError(c)
		return
	}
	resOk := response.Response{Data: *art}
	resOk.Send(c)
}


// @Tags Article
// @Summary 获取文章列表
// @Produce  application/json
// @Success 200 {string} string "{state:{code:0, msg:"成功"},"data":{ }}"
// @Router /api/v1/article/get_article_list [get]
func GetArticleList(c *gin.Context) {
	lo := request.GetLimitOffset(c)
	artist, con,  err := service.GetArticleList(*lo)
	if err != nil {
		res := response.Response{
			Err: err,
			State: response.State{Message: "文章获取失败", Code: errcode.FindError},
		}
		res.SendError(c)
		return
	}
	resOk := response.Response{Data:map[string]interface{}{
		"article_list":artist,
		"limit_offset": lo,
		"count":con,
	}}
	resOk.Send(c)
}


// @Tags Article
// @Summary 创建标签
// @Produce  application/json
// @Success 200 {string} string "{state:{code:0, msg:"成功"},"data":{ }}"
// @Router /api/v1/article/tags/create_tag [post]
func CreateTag (c *gin.Context)  {
	var newtag request.Tag
	if err := c.ShouldBindJSON(&newtag); err != nil {
		res := response.Response{Err: err}
		res.SendInputParameterError(c)
		return
	}

	ar := model.ArticleTags {
		Tag: newtag.Tag,
		TagColor: newtag.TagColor,
		TagDesc: newtag.TagDesc,
	}

	tag, er := service.CreateTag(&ar)
	if er != nil {
		res := response.Response{
			Err: er,
			State: response.State{Message: "标签创建失败", Code: errcode.CreateError},
		}
		res.SendError(c)
		return
	}
	resOk := response.Response{Data: *tag}
	resOk.Send(c)
}



func GetTagList (c *gin.Context) {
	lo := request.GetLimitOffset(c)
	tags, con , err := service.GetTagLis(*lo)
	if err != nil {
		res := response.Response{
			Err: err,
			State: response.State{Message: "标签获取失败", Code: errcode.FindError},
		}
		res.SendError(c)
		return
	}
	resOk := response.Response{Data:map[string]interface{}{
		"tag_list":tags,
		"limit_offset": lo,
		"count":con,
	}}
	resOk.Send(c)
}

File: app/src/api/article.ts
import { Post, confirmStatus, Get } from "@/utils/service";
import type { article, state, articleList, ArticleType } from "@/types/response";
import type {
  articleStorageType,
  CosTempKeyRequest,
  article_req,
  tags,
} from "@/types/request";
import { putObject } from "./cos";
export async function getArticleList(
  limit?: number,
  offset?: number
): Promise<articleList> {
  return new Promise<articleList>((resolve, reject) => {
    Get(`article/get_article_list?limit=${limit}&offset=${offset}`)
      .then((article) => {
        const state = confirmStatus(<state>article.data.state);
        if (state) {
          resolve(article.data.data);
        } else {
          reject(article.data.state);
        }
      })
      .catch(reject);
  });
}

export async function createArticle(
  art: article_req,
  key: Array<string>
): Promise<boolean> {
  return new Promise<boolean>((resolve, reject) => {
    if (art.articleStorageType === articleStorageType.oos) {
      const cosOpt: CosTempKeyRequest = {
        Region: "ap-nanjing",
        Bucket: "bloghtml-1301735126",
        action: "",
      };
      const path = key.concat(art.articleName).join("/");
      putObject(cosOpt, art.articleContext, path, (err, data) => {
        if (err) {
          reject(err);
        } else {
          if (data.statusCode === 200) {
            art.articleUrl = data.Location;
            art.articleContext = "";
            postArticle(art).then(resolve).catch(reject);
          } else {
            reject(data);
          }
        }
      });
    } else {
      postArticle(art).then(resolve).catch(reject);
    }
  });
}

export async function postArticle(art: article_req): Promise<boolean> {
  return new Promise<boolean>((resolve, reject) => {
    Post("article/create_article", art)
      .then((res) => {
        const state = confirmStatus(<state>res.data.state);
        if (state) {
          resolve(res.data.data);
        } else {
          reject(res.data.state);
        }
      })
      .catch(reject);
  });
}

export async function getTagList(limit: number, offset: number): Promise<any> {
  return new Promise<any>((resolve, reject) => {
    Get(`article/tags/get_tags?limit=${limit}&offset=${offset}`)
      .then((res) => {
        const state = confirmStatus(<state>res.data.state);
        if (state) {
          resolve(res.data.data);
        } else {
          reject(res.data.state);
        }
      })
      .catch(reject);
  });
}

export async function createTag(tag: tags): Promise<boolean> {
  return new Promise<boolean>((resolve, reject) => {
    Post("article/tags/create_tag", tag)
      .then((res) => {
        const state = confirmStatus(<state>res.data.state);
        if (state) {
          resolve(res.data.data);
        } else {
          reject(res.data.state);
        }
      })
      .catch(reject);
  });
}

export async function getArticleType(): Promise<Array<ArticleType>> {
  return new Promise<Array<ArticleType>>((resolve, reject) => {
    Get(`article/types/get_types`)
      .then((res) => {
        const state = confirmStatus(<state>res.data.state);
        if (state) {
          resolve(res.data.data);
        } else {
          reject(res.data.state);
        }
      })
      .catch(reject);
  });
}

export async function createArticleType (atype: ArticleType) : Promise<ArticleType> {
  return new Promise<ArticleType>((resolve, reject) => {
    Post('article/types/create_type', atype)
    .then(res => {
       const state = confirmStatus(<state>res.data.state);
        if (state) {
          resolve(res.data.data);
        } else {
          reject(res.data.state);
        }
    })
    .catch(reject)
  })
}
